src/commands/query.js

```diff
@@ -1,92 +1,64 @@
-
-      let config = require(configPath);
-      let scripts = config.developer.scripts;
-      let runScript = {};
-      let dockerArchwayD, archwaydCmd;
-      runScript.raw = scripts.query;
-      if (docker) {
-        dockerArchwayD = commands.ArchwayDocker;
-        archwaydCmd = dockerArchwayD.cmd + ' ' + dockerArchwayD.args.join(" ");
-        runScript.raw = runScript.raw.replace('archwayd', archwaydCmd);
-      }
-      runScript.arr = runScript.raw.split(' ');
-      runScript.cmd = runScript.arr[0];
-      runScript.params = runScript.arr.slice(1);
-
-      // Extend params to include subcommands and args
+  const { chainId, urls: { rpc } = {}, gas = {} } = config.get('network', {});
+  const node = `${rpc.url}:${rpc.port}`;
+  const { address: lastDeployedContract } = config.deployments.findLast('instantiate', chainId) || {};
+  prompts.override({ contract: lastDeployedContract || undefined, ...options });
 
-      // Lvl. 1 cmd
-      runScript.params.push(args.command);
+  const {contract } = await prompts([
+    {
+      type: 'text',
+      name: 'contract',
+      message: chalk`Enter the smart contract address you would like to query from {reset.dim (e.g. "archway1...")}`,
+      validate: value => isArchwayAddress(_.trim(value)) || 'Invalid address',
+      format: value => _.trim(value),
+    },
+  ]);
 
-      // Lvl. 2 cmd
-      if (args.subcommand) {
-        runScript.params.push(args.subcommand);
-      }
 
-      // Address to query
-      // For now defaults to most recent deployment
-      let deployments = config.developer.deployments;
-      for (let i = 0; i < deployments.length; i++) {
-        if (deployments[i].address) {
-          let contract = deployments[i].address;
-          runScript.params.push(contract);
-          // Query to make
-          runScript.params.push(args.query);
 
-          // Additional flags
-          let rpc = config.network.urls.rpc;
-          let node = rpc.url + ':' + rpc.port;
-          let flags = (args.flags) ? args.flags.split(' ') : [];
-          flags.push('--node');
-          flags.push(node);
-          flags.push('--output');
-          flags.push('json');
-
-          const source = spawn(runScript.cmd, runScript.params.concat(flags), { stdio: 'inherit' });
-
-          source.on('error', (err) => {
-            console.log('Error running query', err);
-          });
+  return {
+    contract,
+    args: args || '{}',
+    chainId,
+    node,
+    gas,
+    flags: [...flags], 
+  }
+}
 
-          source.on('close', () => {
-            console.log('\nOk!');
-          })
-          break;
-        }
-      }
-    }
-  });
+async function querySmart(archwayd, options) {
+  const config = await Config.open();
+  const { node, contract, args, ...txOptions } = await parseQueryOptions(config, options);
+  console.info(chalk`Querying smart contract {cyan ${contract}}...`);
+  const  response= await archwayd.query.smartContract(contract, args, { node, ...txOptions });
```

Comment:
Please, fix the spacing. The linter should flag those syntax issues, but if it's not doing that, we should change the rules accordingly.